addons/pos_sale/models/sale_order.py

```diff
@@ -26,6 +27,14 @@ def action_view_pos_order(self):
             'domain': [('id', 'in', linked_orders.ids)],
         }
 
+    @api.depends('order_line', 'amount_total', 'order_line.invoice_lines.move_id.state')
```

Comment:
here it seems you can use "order_line.invoice_lines.parent_state" which is a related stored field